### PR TITLE
[VC] job rest api support vc and add vc label for docker container

### PR DESCRIPTION
**Pull request by wangcan0329**

### CODE REVIEW

File: rest-server/src/templates/yarnContainerScript.mustache [hunk 1]
```diff
@@ -87,6 +87,7 @@ docker run --name $docker_name \
   --volume /tmp/pai-root/alive/$APP_ID:/alive \
   --label PAI_HOSTNAME="$(hostname)" \
   --label PAI_JOB_NAME={{{ jobData.jobName }}} \
+  --label PAI_JOB_VC_NAME={{{ jobData.virtualCluster }}} \
```
Comment (hwuu) [inline rest-server/src/templates/yarnContainerScript.mustache]: `PAI_VC_NAME` is sufficient?

File: rest-server/src/models/job.js [hunk 2]
```diff
@@ -95,6 +95,7 @@ class Job {
               createdTime: frameworkInfo.firstRequestTimestamp || new Date(2018, 1, 1).getTime(),
               completedTime: frameworkInfo.frameworkCompletedTimestamp,
               appExitCode: frameworkInfo.applicationExitCode,
+              virtualCluster: frameworkInfo.queue,
```
Comment (hwuu) [inline rest-server/src/models/job.js]: Please also add 'virtualCluster' field to the result of job details API (GET /jobs/<jobName>).

File: rest-server/src/templates/yarnContainerScript.mustache (hunk 1 above)
Comment (wangcan0329) [reply rest-server/src/templates/yarnContainerScript.mustache]: I think here means this job belong to which virtualCluster, maybe PAI_JOB_VC_NAME will be more precise?

File: rest-server/src/models/job.js (hunk 2 above)
Comment (wangcan0329) [reply rest-server/src/models/job.js]: Sure.

File: rest-server/src/templates/yarnContainerScript.mustache (hunk 1 above)
Comment (hwuu) [reply rest-server/src/templates/yarnContainerScript.mustache]: Ok, it's more precise.

File: rest-server/test/jobDetail.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      .get('/api/v1/jobs/test_job')
+      .end((err, res) => {
+        expect(res, 'status code').to.have.status(200);
+        expect(res, 'json response').be.json;
+        done();
+      });
+  });
+
+  //
+  // Negative cases
```
Comment (hwuu) [inline rest-server/test/jobDetail.js]: Please also add a negative case, as in jobSshInfo.js and jobConfig.js, in which rest-server cannot connect to launcher.

File: rest-server/test/jobDetail.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  //
+  // Positive cases
+  //
+  // GET /api/v1/jobs/test_job
+  it('Case 1 (Positive): should return test_job detail info', (done) => {
+    chai.request(server)
+      .get('/api/v1/jobs/test_job')
+      .end((err, res) => {
+        expect(res, 'status code').to.have.status(200);
+        expect(res, 'json response').be.json;
```
Comment (hwuu) [inline rest-server/test/jobDetail.js]: I think it is better to give a simple check to the content of the json. For example, it should contain a field called 'virtualCluster', etc.

File: rest-server/test/jobDetail.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+                    'taskCreatedTimestamp': 1522559440659,
+                    'taskCompletedTimestamp': 1522560381565,
+                    'taskServiceStatus': {
+                      'serviceVersion': 0
+                    },
+                    'containerId': 'container_e11_1522466352615_0006_01_000002',
+                    'containerHost': '10.151.40.131',
+                    'containerIp': '10.151.40.131',
+                    'containerGpus': 0,
+                    'containerLogHttpAddress': 'http://10.151.40.131:8042/node/containerlogs/container_e11_1522466352615_0006_01_000002/admin/',
```
Comment (hwuu) [inline rest-server/test/jobDetail.js]: It is not necessary to include such details. And you may want to make use of the `frameworkDetailTemplate` defined in `setup.js`.

File: rest-server/test/jobDetail.js (hunk 4 above)
Comment (wangcan0329) [reply rest-server/test/jobDetail.js]: done

File: rest-server/test/jobDetail.js (hunk 3 above)
Comment (wangcan0329) [reply rest-server/test/jobDetail.js]: done

**Pull request merged**